exp/addons/internal/controllers/clusterresourcesetbinding_controller.go

```diff
@@ -117,3 +125,21 @@ func (r *ClusterResourceSetBindingReconciler) clusterToClusterResourceSetBinding
 		},
 	}
 }
+
+func (r *ClusterResourceSetBindingReconciler) updateClusterResourceSetBinding(ctx context.Context, binding *addonsv1.ClusterResourceSetBinding, clusterName string) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
```

Comment:
Let's change to a patchhelper instead of the retry + update pattern to be consistent with the pattern in other controllers.